server/sonar-web/src/main/js/apps/sessions/components/LoginFormContainer.tsx

```diff
@@ -51,14 +59,12 @@ class LoginFormContainer extends React.PureComponent {
   handleSuccessfulLogin = () => {
     const { location } = this.props;
     const queryReturnTo = location.query['return_to'];
-    const returnTo = queryReturnTo ? `${queryReturnTo}${location.hash}` : `${window.baseUrl}/`;
-    window.location = returnTo;
+    const returnTo = queryReturnTo ? `${queryReturnTo}${location.hash}` : `${getBaseUrl()}/`;
+    this.context.router.push(returnTo);
```

Comment:
two problems here:

- first, as you're redirecting with router and both `queryReturnTo` and `getBaseUrl()` are paths with `baseUrl` included, this will lead to the double web context problem (`/sonarqube/sonarqube`)

- second, it's better to fully reload the page after login to re-fetch all data